poly-commitment/src/commitment.rs

```diff
@@ -879,6 +886,35 @@ pub fn inner_prod<F: Field>(xs: &[F], ys: &[F]) -> F {
     res
 }
 
+/// Cast the evaluations in specfic domain size to the smaller domain size.
+///
+/// ## Panics
+///
+/// Panics if `evals_domain_size` is smaller than `target_domain_size`.
+pub fn to_domain<F: FftField>(
+    evals: &Evaluations<F, D<F>>,
+    evals_domain_size: usize,
+    target_domain_size: usize,
+    target_domain: D<F>,
+    shift: usize,
+    constant: Option<F>,
+) -> Evaluations<F, D<F>> {
+    let scale = evals_domain_size / target_domain_size;
+    assert_ne!(
+        scale, 0,
+        "we can't move to a bigger domain without interpolating and reevaluating the polynomial"
+    );
+    let f = |i| {
+        if let Some(cst) = constant {
+            cst + evals.evals[(scale * i + evals_domain_size * shift) % evals.evals.len()]
+        } else {
+            evals.evals[(scale * i + evals_domain_size * shift) % evals.evals.len()]
+        }
+    };
```

Comment:
I think it's better to return a closure here, instead of having the branching be done in the closure unnecessarily